### ISSUE TITLE
Baseline file not generated despite setting `BUNDLE_STATS_BASELINE` environment variable

### ISSUE THREAD

**Issue by carloscuesta** [relative-ci/bundle-stats#4181]
## Description

Hey! 👋🏼 

I have been using `bundle-stats-webpack-plugin` for a while now and it has worked without any issues, thanks for this great tool ❤️ 

However since `4.7.1` I noticed that our `baseline` stats file is not being generated as it was. Last version that was working is `4.6.3`.

I'm setting the `BUNDLE_STATS_BASELINE` environment variable  to `BUNDLE_STATS_BASELINE=true` however I still get the `Baseline stats file is missing, see "baseline" option.` warning in the console output.

### Setup

Our setup, is quite simple in terms of code, we import the plugin conditionally and pass two options: `outDir` and `json`:

<img width="1055" alt="Screenshot 2024-01-22 at 10 28 27" src="https://github.com/relative-ci/bundle-stats/assets/7629661/f0a0d8c0-e55e-4c77-ae61-59efced08e65">

When trying to build the app locally and trying to generate the bundle-stats, I can see the `Baseline stats file` missing log:

![Local](https://github.com/relative-ci/bundle-stats/assets/7629661/ed843ee3-08b4-47c2-a18e-42ebf303f582)

The same thing happens when I run the build from CI (GitHub Actions): 

![CI](https://github.com/relative-ci/bundle-stats/assets/7629661/ebde1ec2-88b7-46b8-972e-1eb1e94ecb21)

### Potential problem?

I was taking a look into it and saw that when upgrading from `4.6.3` to `4.9.x` there was a modification here https://github.com/relative-ci/bundle-stats/pull/3844 that changes the behaviour of the environment variable, wondering if that can be the reason why it's not working 🤔 👀 

Thanks for the help! 🙏🏼

**Comment by vio** [comment]
@carloscuesta thanks for creating the issue! 

It is most likely a regression if it stopped working after the upgrade. 

Just in case, can you please confirm if the baseline file is available for your build process? By default, the file is saved on `node_modules/.cache/bundle-stats/baseline.json`

**Comment by carloscuesta** [comment]
Hey @vio thanks for the quick response 🙌🏼 

> It is most likely a regression if it stopped working after the upgrade.

Could be! 

> Just in case, can you please confirm if the baseline file is available for your build process? By default, the file is saved on node_modules/.cache/bundle-stats/baseline.json

There's no `baseline.json` stats file on the `node_modules/.cache/bundle-stats` in fact the directory is not there either!

**Comment by savichev-igor** [comment]
I have a same problem and fast fix for right way for baseline.json:
```
new BundleStatsWebpackPlugin({
      baselineFilepath: './node_modules/.cache/bundle-stats/baseline.json', // set this option
      outDir: `reports`,
      json: true,
}),
```

**Comment by vio** [comment]
Hi @carloscuesta! I ran a bunch of tests, and I could not replicate the issue. The related changes are unit-tested, so I suspect an issue with the env variable.

I released a new version ([`v4.9.2-beta.1`](https://github.com/relative-ci/bundle-stats/pull/4185)) that improves baseline reading/writing logging. If you get the chance, you can try the following tests:

1. Clear your `node_modules/.cache/bundle-stats`
2. Run the build task with `BUNDLE_STATS_BASELINE=true` - should save the current webpack stats as a baseline and produce logs similar to these:

```shell
Reading baseline data from ../node_modules/.cache/bundle-stats/baseline.json. Baseline stats file is missing, see "baseline" option.
Writing baseline data to ../node_modules/.cache/bundle-stats/baseline.json
Bundle Size — 1012.57KiB (+100%).
````

2. Run the build command again - should use the previously saved baseline to compare and create the report

```shell
Reading baseline data from ../node_modules/.cache/bundle-stats/baseline.json.
Bundle Size — 1012.57KiB (0%).
```

If the compare failed with missing baseline, you can try to run the tests again with `baseline` option set directly on the webpack plugin options: https://github.com/relative-ci/bundle-stats/tree/master/packages/webpack-plugin#bundlestatswebpackpluginoptions

**Comment by carloscuesta** [comment]
Hey @vio thanks for the quick reply sharing my results here:

1. Clean all the `.cache/bundle-stats` folders

2. Install the dependency:

`$ head node_modules/bundle-stats-webpack-plugin/package.json`

```
{
  "name": "bundle-stats-webpack-plugin",
  "version": "4.9.2-beta.1",
  "description": "In-depth bundle analyzer for webpack(bundle size, assets, modules, packages)",
  "main": "lib/index.js",
  "types": "typings.d.ts",
  "directories": {
    "lib": "lib"
  },
  "engines": {
```

3. Run the build command:

![build](https://github.com/relative-ci/bundle-stats/assets/7629661/bb6786f9-2b4d-46d5-ac6d-5c986956078b)

4. Run the build command again without the BASELINE environment variable to use compare mode:

![build-compare)](https://github.com/relative-ci/bundle-stats/assets/7629661/0239b22d-e0de-4abe-af6a-b091aa29d5e7)

I still see the warning missing on the logs, however I was able to find a `baseline.json` file in the tree which is located in:

```
./../node_modules/.cache/bundle-stats/baseline.json
``` 

Which is a different directory than the one the plugin is reporting in the logs 

> If the compare failed with missing baseline, you can try to run the tests again with baseline option set directly on the webpack plugin options: https://github.com/relative-ci/bundle-stats/tree/master/packages/webpack-plugin#bundlestatswebpackpluginoptions

Same results! Tried adding `baseline: true` to the configuration and then removing it for the compare build. Looks like the value of the environment variable is being properly detected but there's something fishy with where the plugin is trying to read the `baseline.json` file

**Comment by carloscuesta** [comment]
Found something!!

I removed the `outDir` option from the plugin and I can see in the logs that now the baseline file is located and everything seems to be working as expected:

```diff
  new BundleStatsWebpackPlugin({
-    outDir: 'bundle-stats',
    json: true,
  })
```

**Build with baseline:**

```
> NODE_ENV=production WEBPACK_BUNDLE_STATS=1 BUNDLE_STATS_BASELINE=true pnpm run build-app

> webpack --config webpack.prod.js

Reading baseline data from ../node_modules/.cache/bundle-stats/baseline.json.
Writing baseline data to ../node_modules/.cache/bundle-stats/baseline.json
```

**Build with compare:**

```
> NODE_ENV=production WEBPACK_BUNDLE_STATS=1 pnpm run build-app

> @goldfish/servers@9.0.0 build-app /Users/carloscuesta/Developer/goldfish/packages/servers
> webpack --config webpack.prod.js

Reading baseline data from ../node_modules/.cache/bundle-stats/baseline.json.
Bundle Size — 13.16MiB (0%).
```

**Comment by carloscuesta** [comment]
Found something!!

I removed the `outDir` option from the plugin and I can see in the logs that now the baseline file is located and everything seems to be working as expected:

```diff
  new BundleStatsWebpackPlugin({
-    outDir: 'bundle-stats',
    json: true,
  })
```

**Build with baseline:**

```
> NODE_ENV=production WEBPACK_BUNDLE_STATS=1 BUNDLE_STATS_BASELINE=true pnpm run build-app

> webpack --config webpack.prod.js

Reading baseline data from ../node_modules/.cache/bundle-stats/baseline.json.
Writing baseline data to ../node_modules/.cache/bundle-stats/baseline.json
```

**Build with compare:**

```
> NODE_ENV=production WEBPACK_BUNDLE_STATS=1 pnpm run build-app

Reading baseline data from ../node_modules/.cache/bundle-stats/baseline.json.
Bundle Size — 13.16MiB (0%).
```

**Comment by vio** [comment]
Great find, thanks for taking the time! It looks like we don't consider `outDir` correctly when generating the relative path to the webpack output path. The baseline is saved but in the parent folder.

A fix was released in `v4.9.2-beta.3`. If you try it, let me know how it goes.

**Comment by carloscuesta** [comment]
`v4.9.2-beta.3` works like a charm! 🥳 

```
> NODE_ENV=production WEBPACK_BUNDLE_STATS=1 BUNDLE_STATS_BASELINE=true pnpm run build-app

> webpack --config webpack.prod.js
Reading baseline data from ../node_modules/.cache/bundle-stats/baseline.json.
Writing baseline data to ../node_modules/.cache/bundle-stats/baseline.json
```


```
> N26_SERVER=banking NODE_ENV=production WEBPACK_BUNDLE_STATS=1 pnpm run build-app
> webpack --config webpack.prod.js
Reading baseline data from ../node_modules/.cache/bundle-stats/baseline.json.
Bundle Size — 13.16MiB (0%).
```

Thank you very much for the help ❤️

**Comment by carloscuesta** [comment]
`v4.9.2-beta.3` works like a charm! 🥳 

```
> NODE_ENV=production WEBPACK_BUNDLE_STATS=1 BUNDLE_STATS_BASELINE=true pnpm run build-app
> webpack --config webpack.prod.js
Reading baseline data from ../node_modules/.cache/bundle-stats/baseline.json.
Writing baseline data to ../node_modules/.cache/bundle-stats/baseline.json
```


```
> NODE_ENV=production WEBPACK_BUNDLE_STATS=1 pnpm run build-app
> webpack --config webpack.prod.js
Reading baseline data from ../node_modules/.cache/bundle-stats/baseline.json.
Bundle Size — 13.16MiB (0%).
```

Thank you very much for the help ❤️

**Comment by vio** [comment]
@carloscuesta i released the fix on [`v4.9.2`](https://github.com/relative-ci/bundle-stats/releases/tag/v4.9.2). 

Thanks for your help! 🙇
